src/main/java/io/vertx/json/schema/common/ItemsValidatorFactory.java

```diff
@@ -41,10 +42,13 @@ public ItemsValidator(MutableStateValidator parent) {
     public void validateSync(ValidatorContext context, Object in) throws ValidationException, NoSyncValidationException {
       this.checkSync();
       if (in instanceof JsonArray) {
-        JsonArray arr = (JsonArray) in;
+        in = ((JsonArray) in).getList();
+      }
+      if (in instanceof List) {
+        List<?> arr = (List<?>) in;
         for (int i = 0; i < arr.size(); i++) {
           context.markEvaluatedItem(i);
-          schema.validateSync(context.lowerLevelContext(), arr.getValue(i));
+          schema.validateSync(context.lowerLevelContext(), JsonUtil.wrapJsonValue(arr.get(i)));
```

Comment:
Do we really need the wrap when going down the structure? If the structure is something like `List<List<Integer>>`, it should work anyway, right?